build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -616,34 +613,27 @@ class _SingleBuild {
     int actionNum,
     PostBuildAction action,
   ) async {
-    var anchorNodes =
-        _assetGraph.packageNodes(action.package).where((node) {
-          if (node is PostProcessAnchorNode && node.actionNumber == actionNum) {
-            var inputNode = _assetGraph.get(node.primaryInput);
-            if (inputNode is SourceAssetNode) {
-              return true;
-            } else if (inputNode is GeneratedAssetNode) {
-              return inputNode.wasOutput &&
-                  !inputNode.isFailure &&
-                  inputNode.state == NodeState.upToDate;
-            }
-          }
-          return false;
-        }).cast<PostProcessAnchorNode>();
-    var outputLists = await Future.wait(
-      anchorNodes.map(
-        (anchorNode) => _runPostProcessBuilderForAnchor(
-          phaseNum,
-          actionNum,
-          action.builder,
-          anchorNode,
-        ),
-      ),
-    );
-    return outputLists.fold<List<AssetId>>(
-      <AssetId>[],
-      (combined, next) => combined..addAll(next),
-    );
+    final outputs = <AssetId>[];
+    for (final node in _assetGraph.packageNodes(action.package).toList()) {
```

Comment:
if the `.toList()` is needed you should probably pass `growable: false`.

Reply:
Done.